### PR TITLE
Loosen conditions on MemoryPacking

### LINKED ISSUE #2196: MemoryPacking should support running with bulk-memory enabled
As long as the module still uses active segments, e.g. in a single-threaded application. MemoryPacking could theoretically also be taught to reason about passive segments and `memory.init` instructions, but that is low priority because the problem can be better addressed in the linker.

See https://github.com/emscripten-core/emscripten/issues/8899 for context.

### PR DESCRIPTION
**Pull request by tlively**
Allow MemoryPacking to run when there are no passive segments, even if
bulk memory is enabled.

Fixes #2196.

### CODE REVIEW

File: src/passes/MemoryPacking.cpp [hunk 1]
```diff
@@ -28,12 +28,19 @@ struct MemoryPacking : public Pass {
   bool modifiesBinaryenIR() override { return false; }
 
   void run(PassRunner* runner, Module* module) override {
-    // Conservatively refuse to change segments if bulk memory is enabled to
-    // avoid invalidating segment indices or segment contents referenced from
+    if (!module->memory.exists)
+      return;
```
Comment (kripken) [inline src/passes/MemoryPacking.cpp]: `{ }` if not on the same line.

@aheejin didn't we have clang-tidy check for that?

Comment (aheejin) [reply src/passes/MemoryPacking.cpp]: We do. Travis CI is pending now, but once it runs it will error out.

**Comment by tlively** [comment]
The test must have been a holdover from when all features were enabled by default. I guess we should go through and make sure there aren't any others.

File: src/passes/MemoryPacking.cpp (hunk 1 above)
Comment (kripken) [reply src/passes/MemoryPacking.cpp]: Thanks, and sorry I didn't notice it was still pending.

**Comment by kripken** [comment]
lgtm

**Pull request merged**